sycl/plugins/level_zero/pi_level0.cpp

```diff
@@ -2291,9 +2368,26 @@ pi_result piEventsWait(pi_uint32 NumEvents, const pi_event *EventList) {
     // NOTE: we are destroying associated command lists here to free
     // resources sooner in case RT is not calling piEventRelease soon enough.
     if (EventList[I]->ZeCommandList) {
-      // Event has been signaled: Destroy the command list associated with the
-      // call that generated the event.
-      ZE_CALL(zeCommandListDestroy(EventList[I]->ZeCommandList));
+      // Event has been signaled: If the fence for the associated command list
```

Comment:
Please re-use same code in `piEventRelease`

Reply:
Updated to use the new resetCommandListFenceEntry function.